src/types/serialized-type.ts

```diff
@@ -1,6 +1,96 @@
 import { BytesList } from "../serdes/binary-serializer";
 const { bytesToHex, slice } = require("../utils/bytes-utils");
 
+/**
+ * The base class for all binary-codec types
+ */
+class SerializedTypeClass {
```

Comment:
This one needs some explaining. All classes that use `makeClass` still require `SerializedType` (the object) for mix-ins/inheritance. 

So I defined a `SerializedTypeClass` to be used with the code that is being refactored in this PR, and left `SerializedType` alone so that it wouldn't affect the existing functionality of types not being refactored. 

Same with `ComparableClass` and `Comparable`

Once I'm done, I will delete the old `Comparable` and `SerializedType`, and will rename the classes`ComparableClass`->`Comparable` and `SerializedTypeClass`->`SerializedType`